explainer-server/app/controllers/ApiController.scala

```diff
@@ -76,6 +76,50 @@ class ExplainerApiImpl(
     explainerToPublish.map(CsAtom.atomToCsAtom)
   }
 
+  override def addTagToExplainer(explainerId: String, tagId: String): Future[CsAtom] = {
+    for {
+      csatom <- explainerDB.load(explainerId).map(CsAtom.atomToCsAtom)
+    }yield{
+      val csexplaineratom: CsExplainerAtom = csatom.data
+      val tags: Option[List[String]] = csexplaineratom.tags
+      val newtags: Option[List[String]] = tags match {
+        case None => Some(List(tagId))
+        case Some(list) => Some( (tagId :: list).distinct.sorted )
+      }
+      val newCsExplainerAtom: CsExplainerAtom = CsExplainerAtom(csexplaineratom.title, csexplaineratom.body, csexplaineratom.displayType, newtags)
+      val newCsAtom: CsAtom = CsAtom(csatom.id, newCsExplainerAtom, csatom.contentChangeDetails)
+      explainerDB.store(CsAtom.csAtomToAtom(newCsAtom))
+      explainerStore.updateLastModified(explainerId, user)
+    }
+    load(explainerId)
+  }
```

Comment:
One way of simplifying this would be to use the Scala copy function, which copies something, allowing you to replace just specific fields. The thrift conversions make it still a bit messy but saves you from specifying things that haven't changed (on lines 89,90):

e.g.
`    explainerDB.load(explainerId).map(explainer => {
      val updatedExplainer = explainer.copy(
        data = AtomData.Explainer(explainer.tdata.copy(
          tags =Some((tagId +: explainer.tdata.tags.getOrElse(List())).distinct.sorted)
      )))
      explainerDB.store(explainer)
      CsAtom.atomToCsAtom(updatedExplainer)
    })`